doc/upgrading.md

```diff
@@ -17,6 +18,24 @@ See the [NEWS](../NEWS.md) summary of changes for each version.
 * [Version 25](#version-25)
 * [Version 20](#version-20)
 
+## Version 32.0
+
+* [RMB-609](https://issues.folio.org/browse/RMB-609) Upgrade to Vert.x 4:
+  * Replace deprecated `io.vertx.core.logging.Logger` by
+    `org.apache.logging.log4j.Logger`, for example
+    `private static final Logger LOGGER = LogManager.getLogger(Foo.class)`
+    (using `getLogger()` without argument requires Multi-Release for
+    stack walking, see below).
+  * Replace `HttpClient` `.getAbs`, `.postAbs`, `.putAbs`, `.deleteAbs` by
+    `WebClient.requestAbs(HttpMethod method, ...)`.
+  * Replace `Future.setHandler` by `Future.onComplete`.
+  * `Future` has been split into `Future` and `Promise`, see
+    [Futurisation in Vert.x 4](futurisation.md).
+  * In order to have structured logging for tests RMB based modules should copy [log4j2-test.properties](https://github.com/folio-org/raml-module-builder/blob/master/domain-models-runtime/src/test/resources/log4j2-test.properties) into their `/src/test/resources` directory
+* [OKAPI-943](https://issues.folio.org/browse/OKAPI-943) When calling `.any`, `.all` or `.join`
+  replace `io.vertx.core.CompositeFuture` by `org.folio.okapi.common.GenericCompositeFuture`,
+  replace raw type by actual type and remove `@SuppressWarnings("rawtypes")`.
+
```

Comment:
The upgrading notes should mention that ClientGenerator functions has different responseHandler. From 
``
  Handler<HttpClientResponse>
``

to 
``
  Handler<AsyncResult<HttpResponse<Buffer>>>
``